app/views/handlers/primary_person_list_collector.py

```diff
@@ -35,8 +35,6 @@ def handle_post(self, form):
             self._primary_person_id = self.questionnaire_store_updater.add_primary_person(
                 self.rendered_block['for_list']
             )
-            # To ensure answering 'No' doesn't allow the user to skip ahead.
-            self.questionnaire_store_updater.remove_completed_location()
```

Comment:
This has been removed because if you answer `Yes` to the primary person question, then click Previous, then repost the `Yes` answer, then not submit the name question, the primary person list collector block is never marked as complete which causes can cause routing issues.

This line was added to fix another issue, where you answered `No` but then changed the answer to `Yes`.

However, the latter is much less likely to happen therefore, for now, after speaking with @ajmaddaford and @bitdivision, I've decided to make it less broken?

Happy to discuss in IRL if the above is unclear.